### PR TITLE
[improve] Check the topic exists from cache

### PR DESCRIPTION
**Pull request by Demogorgon314**
### Motivation

Using the admin API to check if the "topic" exists is a heavy operation, we should use the metadata cache to check if the topic exists.

### Modifications

Check the topic exists from cache

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopBrokerLookupManager.java [hunk 1]
```diff
@@ -122,21 +120,31 @@ private CompletableFuture<InetSocketAddress> lookupBroker(final String topic) {
...
+                .whenComplete((metadata, ex) -> {
+                    if (metadata.partitions == 0) {
+                        internalCheckTopicExists(topicName).thenAccept(future::complete)
+                                .exceptionally(throwable -> {
+                                    log.error("Check topic exists has exception.", throwable);
+                                    future.complete(true);
+                                    return null;
+                                });
+                    }
                     if (ex != null) {
```
Comment (BewareMyPower) [inline kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopBrokerLookupManager.java:134]: We should check `if (ex != null)` at the beginning of the callback.

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopBrokerLookupManager.java [hunk 2]
```diff
@@ -122,21 +120,31 @@ private CompletableFuture<InetSocketAddress> lookupBroker(final String topic) {
...
-        this.adminClient.topics().getPartitionedTopicMetadataAsync(topic)
-                .whenComplete((__, ex) -> {
+        TopicName topicName = TopicName.get(topic);
+        this.pulsar.getBrokerService().fetchPartitionedTopicMetadataAsync(TopicName.get(topic))
+                .whenComplete((metadata, ex) -> {
+                    if (metadata.partitions == 0) {
+                        internalCheckTopicExists(topicName).thenAccept(future::complete)
+                                .exceptionally(throwable -> {
+                                    log.error("Check topic exists has exception.", throwable);
+                                    future.complete(true);
```
Comment (BewareMyPower) [inline kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopBrokerLookupManager.java:130]: Do you intend to ignore the exception?

Comment (Demogorgon314) [reply kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopBrokerLookupManager.java:135]: Yes, if it has an exception, we should let the `TransactionMarkerChannelManager` retry.

**Review by BewareMyPower — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]
LGTM

**Pull request merged**